src/ai/backend/web/proxy.py

```diff
@@ -127,22 +127,33 @@ async def close_upstream(self) -> None:
             await self.up_conn.close()
 
 
-async def decrypt_payload(request: web.Request) -> str:
-    if not request.content:
-        return ""
-    config = request.app["config"]
-    scheme = config["service"]["force_endpoint_protocol"]
-    if scheme is None:
-        scheme = request.scheme
-    api_endpoint = f"{scheme}://{request.host}"
-    payload = await request.text()
-    iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
-    key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + iv + iv)[0:32]
-    crypt = AES.new(bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(iv, encoding="utf8"))
-    b64p = base64.b64decode(real_payload)
-    dec = unpad(crypt.decrypt(bytes(b64p)), 16)
-    result = dec.decode("UTF-8")
-    return result
+@web.middleware
+async def decrypt_payload(request: web.Request, handler) -> web.StreamResponse:
+    request_headers = extra_config_headers.check(request.headers)
+    secure_context = request_headers.get("X-BackendAI-Encoded", None)
+    if secure_context:
+        if not request.content:
+            request.app["config"]["payload"] = ""
+            return await handler(request)
+        config = request.app["config"]
+        scheme = config["service"]["force_endpoint_protocol"]
+        if scheme is None:
+            scheme = request.scheme
+        api_endpoint = f"{scheme}://{request.host}"
+        payload = await request.text()
+        iv, real_payload = payload.split(":")  # Extract initial vector and actual payload
+        key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + iv + iv)[0:32]
+        crypt = AES.new(bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(iv, encoding="utf8"))
+        b64p = base64.b64decode(real_payload)
+        dec = unpad(crypt.decrypt(bytes(b64p)), 16)
+        result = dec.decode("UTF-8")
+        if not result:
+            request.app["config"]["payload"] = {}
+        else:
+            request.app["config"]["payload"] = json.loads(result)
+    else:
+        request.app["config"]["creds"] = request.json()
```

Comment:
`request.json()` returns Coroutine, not `dict` (ref: https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.BaseRequest.json). You need to add `await` to retrieve actual `dict` data.
```suggestion
        request.app["config"]["creds"] = await request.json()
```

Reply:
When I add await I get that error.
![image](https://user-images.githubusercontent.com/39542923/183004214-77f27275-173f-4ca2-ba6b-a7f66a34dbde.png)

Reply:
Looks like it's not issue of `await`; it's more like unnoticed bug being revealed. What `aiohttp` complaing about is that request body is not a json. Current implementation makes `decrypt_payload()` middleware to do its job on all the requests coming into webserver. And since your middleware is trying to treat all request bodies (which aren't empty) as json, when a request which isn't json (maybe graphql requests?), boom. It'll fail like this.    
IMO there is no reason to try to decode & re-encode request body as json. Let's just keep it as `str`.     
Also, dict key which holds decrypted payload should be some other self-explanatory name, not `creds`. 
```suggestion
        request["payload"] = request
```